usual/crypto/md5.h

```diff
@@ -33,12 +33,31 @@
 /** Result length for MD5 */
 #define MD5_DIGEST_LENGTH	16
 
+#ifndef OPENSSL_API_COMPAT
 /** MD5 state */
 struct md5_ctx {
 	uint64_t nbytes;
 	uint32_t a, b, c, d;
 	uint32_t buf[16];
 };
+#else
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <usual/logging.h>
+
+/* Set of error states */
+typedef enum md5_errno
+{
+	MD5_ERROR_NONE = 0,
+	MD5_ERROR_OPENSSL
+} md5_errno;
+
+struct md5_ctx {
+	md5_errno error;
```

Comment:
This field seems unnecessary, a NULL or non-NULL errorreason field gives the same information.